Topics/Software_Engineering/Ethics_In_Software_Engineering.md

```diff
@@ -0,0 +1,112 @@
+
+### Sephora
+**Violation of Consent**
+
+The problem with the software design of multi-million dollar company Sephora was a concept commonly referred to in conversations surrounding ethics: "opt in vs. opt out". What this means is that, generally, if you offer a user a product or service, or wish to gather their consent for data collection, for example, they must be opted out by default, not opted in. That is, if your website has a check box which the user needs to have selected in order for you to use their data, that check box must be unselected by default.
+
+In 2022, Sephora was found guilty of doing the opposite, opting users in to having their data sold by default. This resulted in a $1.2 million fine as an enforcement of the California Consumer Privacy Act.
+
+### ChatGPT
+**Violation of Security**
+
+On March 24th, 2023, it was revealed that Open AI's system had an easily exploitable vulnerability, allowing users to, through an open-source library called Redis, see the chat histories of other users. This same vulnerability was also responsible for allowing users to view payment information of other users, as well as their full names, email addresses and payment addresses. 
+
+Open AI was thankfully able to detect this issue and take ChatGPT offline before it could impact any more than 1% of their users. However, with a large user base, 1% still implies many affected people, and this raises concerns for future problems as their user base continues to grow.
+
+### Amazon
+**Biased Algorithm**
+
+Another case of unethical computing comes from a hiring algorithm scrapped by Amazon in 2015. Initially created to remove bias that recruiters may have towards candidates, Amazon's algorithm was unfortunatley proved to be just as biased if not more so. Specifically, the algorithm was revealed to have a negative bias towards female candidates, flagging words like "women's" when reveiwing resumes. The reason this happened was because Amazon used a biased dataset to train the algorithm - it was trained using a set of the past 10 years' worth of Amazon resumes, most of the approved ones being from male candidates. Hence, the algorithm "learned" that the ideal Amazon candidate must be male. 
+
+Amazon should have hand-picked the data they trained the algorithm with. Although this would take more time, providing the algorithm with unbiased data, in which the only differences between acceptable and unacceptable resumes are in their merit, would have reduced the likelihood of such discimination occurring. 
+
+## Software Engineering Code of Ethics
+Now that you have some insights into what you shouldn't do, the Software Enginnering Code of Ethics details how you should act when making decisions regarding the ethical designs of your work.
+
+The 8 principles of this code of ethics are as follows:
+* Public: Software engineers shall act consistently with the public interest.
```

Comment:
I would recommend to emphasize the subtitle of each entry, like using single asterisks or different colors to make it better for reader's experience